gc/base/MemoryPoolAddressOrderedListBase.hpp

```diff
@@ -48,6 +48,7 @@ class MM_MemoryPoolAddressOrderedListBase : public MM_MemoryPool
 	MM_HeapLinkedFreeHeader ** _referenceHeapFreeList;
 	MM_SweepPoolState* _sweepPoolState;	/**< GC Sweep Pool State */
 	MM_SweepPoolManagerAddressOrderedListBase* _sweepPoolManager;		/**< pointer to SweepPoolManager class */
+	MM_HeapLinkedFreeHeader *_lastFreeEntry;							/**< address of the last free entry in the pool; valid after compact; NOT maintained during allocation */ 
```

Comment:
agreed, this is a compromise between the correct solution and not too much changes within this PR